From: Sam Ravnborg <srn@skov.dk>
Date: Wed, 12 Oct 2016 19:19:32 +0200
Subject: [PATCH] devicetree: restructure skov files and add support for lt6
 display

- All skov device trees now have "skov" in their name
- Naming scheme simplified
- "compatible" updated to follow filename
- Bord names updated
- Added support the the 6L display
- Added panel label to imx6qdl-skov-cpu so we could rewrite the
  compatible property

Signed-off-by: Sam Ravnborg <srn@skov.dk>
---
 arch/arm/boot/dts/Makefile                         |  5 ++-
 ...skov-imx6-revC.dts => imx6dl-skov-revc-lt2.dts} |  8 ++---
 arch/arm/boot/dts/imx6dl-skov-revc-lt6.dts         | 27 +++++++++++++++
 ...-skov-imx6-revC.dts => imx6q-skov-revc-lt2.dts} |  8 ++---
 arch/arm/boot/dts/imx6q-skov-revc-lt6.dts          | 40 ++++++++++++++++++++++
 ...mx6qdl-cpuC.dtsi => imx6qdl-skov-cpu-revc.dtsi} |  0
 .../{imx6qdl-cpu.dtsi => imx6qdl-skov-cpu.dtsi}    |  2 +-
 7 files changed, 80 insertions(+), 10 deletions(-)
 rename arch/arm/boot/dts/{imx6dl-skov-imx6-revC.dts => imx6dl-skov-revc-lt2.dts} (73%)
 create mode 100644 arch/arm/boot/dts/imx6dl-skov-revc-lt6.dts
 rename arch/arm/boot/dts/{imx6q-skov-imx6-revC.dts => imx6q-skov-revc-lt2.dts} (78%)
 create mode 100644 arch/arm/boot/dts/imx6q-skov-revc-lt6.dts
 rename arch/arm/boot/dts/{imx6qdl-cpuC.dtsi => imx6qdl-skov-cpu-revc.dtsi} (100%)
 rename arch/arm/boot/dts/{imx6qdl-cpu.dtsi => imx6qdl-skov-cpu.dtsi} (99%)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index b7c7cd9a72e6..22632ad76f05 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -302,6 +302,8 @@ dtb-$(CONFIG_SOC_IMX6Q) += \
 	imx6dl-sabreauto.dtb \
 	imx6dl-sabrelite.dtb \
 	imx6dl-sabresd.dtb \
+	imx6dl-skov-revc-lt2.dtb \
+	imx6dl-skov-revc-lt6.dtb \
 	imx6dl-tx6dl-comtft.dtb \
 	imx6dl-tx6u-801x.dtb \
 	imx6dl-tx6u-811x.dtb \
diff --git a/arch/arm/boot/dts/imx6dl-skov-imx6-revC.dts b/arch/arm/boot/dts/imx6dl-skov-revc-lt2.dts
similarity index 73%
rename from arch/arm/boot/dts/imx6dl-skov-imx6-revC.dts
rename to arch/arm/boot/dts/imx6dl-skov-revc-lt2.dts
index a7333f06f932..f987973bc7f6 100644
--- a/arch/arm/boot/dts/imx6dl-skov-imx6-revC.dts
+++ b/arch/arm/boot/dts/imx6dl-skov-revc-lt2.dts
@@ -14,10 +14,10 @@
 /dts-v1/;
 #include <dt-bindings/interrupt-controller/irq.h>
 #include "imx6dl.dtsi"
-#include "imx6qdl-cpu.dtsi"
-#include "imx6qdl-cpuC.dtsi"
+#include "imx6qdl-skov-cpu.dtsi"
+#include "imx6qdl-skov-cpu-revc.dtsi"
 
 / {
-	model = "Freescale i.MX6 SKOV Solo Board";
-	compatible = "skov,imx6-imxdl-revC", "skov,imx6", "fsl,imx6dl";
+	model = "SKOV IMX6 CPU SoloCore";
+	compatible = "skov,imx6dl-skov-revc-lt2", "skov,imx6", "fsl,imx6dl";
 };
diff --git a/arch/arm/boot/dts/imx6dl-skov-revc-lt6.dts b/arch/arm/boot/dts/imx6dl-skov-revc-lt6.dts
new file mode 100644
index 000000000000..cc278ac3f435
--- /dev/null
+++ b/arch/arm/boot/dts/imx6dl-skov-revc-lt6.dts
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2014 Rose Technology
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+#include <dt-bindings/interrupt-controller/irq.h>
+#include "imx6dl.dtsi"
+#include "imx6qdl-skov-cpu.dtsi"
+#include "imx6qdl-skov-cpu-revc.dtsi"
+
+/ {
+	model = "SKOV IMX6 CPU SoloCore";
+	compatible = "skov,imx6dl-skov-revc-lt6", "skov,imx6", "fsl,imx6dl";
+};
+
+&panel {
+	compatible = "logictechnologies,lttd800480070-l6wh-rt";
+};
diff --git a/arch/arm/boot/dts/imx6q-skov-imx6-revC.dts b/arch/arm/boot/dts/imx6q-skov-revc-lt2.dts
similarity index 78%
rename from arch/arm/boot/dts/imx6q-skov-imx6-revC.dts
rename to arch/arm/boot/dts/imx6q-skov-revc-lt2.dts
index 81730c79fd0c..7132c1fd6094 100644
--- a/arch/arm/boot/dts/imx6q-skov-imx6-revC.dts
+++ b/arch/arm/boot/dts/imx6q-skov-revc-lt2.dts
@@ -14,12 +14,12 @@
 /dts-v1/;
 #include <dt-bindings/interrupt-controller/irq.h>
 #include "imx6q.dtsi"
-#include "imx6qdl-cpu.dtsi"
-#include "imx6qdl-cpuC.dtsi"
+#include "imx6qdl-skov-cpu.dtsi"
+#include "imx6qdl-skov-cpu-revc.dtsi"
 
 / {
-	model = "Freescale i.MX6 SKOV Quad Board";
-	compatible = "skov,imx6-imxq-revC", "skov,imx6", "fsl,imx6q";
+	model = "SKOV IMX6 CPU QuadCore";
+	compatible = "skov,imx6q-skov-revc-lt2", "skov,imx6", "fsl,imx6q";
 };
 
 &sata {
diff --git a/arch/arm/boot/dts/imx6q-skov-revc-lt6.dts b/arch/arm/boot/dts/imx6q-skov-revc-lt6.dts
new file mode 100644
index 000000000000..07dd8bea7f24
--- /dev/null
+++ b/arch/arm/boot/dts/imx6q-skov-revc-lt6.dts
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2014 Rose Technology
+ * Copyright 2012 Freescale Semiconductor, Inc.
+ * Copyright 2011 Linaro Ltd.
+ *
+ * The code contained herein is licensed under the GNU General Public
+ * License. You may obtain a copy of the GNU General Public License
+ * Version 2 or later at the following locations:
+ *
+ * http://www.opensource.org/licenses/gpl-license.html
+ * http://www.gnu.org/copyleft/gpl.html
+ */
+
+/dts-v1/;
+#include <dt-bindings/interrupt-controller/irq.h>
+#include "imx6q.dtsi"
+#include "imx6qdl-skov-cpu.dtsi"
+#include "imx6qdl-skov-cpu-revc.dtsi"
+
+/ {
+	model = "SKOV IMX6 CPU QuadCore";
+	compatible = "skov,imx6q-skov-revc-lt6", "skov,imx6", "fsl,imx6q";
+};
+
+&sata {
+	status = "disabled";
+};
+
+&i2c2 {
+	status = "okay";
+};
+
+&hdmi {
+	status = "okay";
+	ddc-i2c-bus = <&i2c2>;
+};
+
+&panel {
+	compatible = "logictechnologies,lttd800480070-l6wh-rt";
+};
diff --git a/arch/arm/boot/dts/imx6qdl-cpuC.dtsi b/arch/arm/boot/dts/imx6qdl-skov-cpu-revc.dtsi
similarity index 100%
rename from arch/arm/boot/dts/imx6qdl-cpuC.dtsi
rename to arch/arm/boot/dts/imx6qdl-skov-cpu-revc.dtsi
diff --git a/arch/arm/boot/dts/imx6qdl-cpu.dtsi b/arch/arm/boot/dts/imx6qdl-skov-cpu.dtsi
similarity index 99%
rename from arch/arm/boot/dts/imx6qdl-cpu.dtsi
rename to arch/arm/boot/dts/imx6qdl-skov-cpu.dtsi
index f6d8d65252dc..73a118bdedf1 100644
--- a/arch/arm/boot/dts/imx6qdl-cpu.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-skov-cpu.dtsi
@@ -236,7 +236,7 @@
 		status = "okay";
 	};
 
-	panel {
+	panel: panel {
 		compatible = "logictechnologies,lttd800480070-l2rt";
 		backlight = <&backlight_lcd>;
 		power-supply = <&reg_3p3v>;
